core-rust/state-manager/src/jni/p2p/high_priority_peers.rs

```diff
@@ -0,0 +1,116 @@
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+use jni::JNIEnv;
+use jni::objects::{JClass, JObject};
+use jni::sys::jbyteArray;
+
+use node_common::java::*;
+
+use crate::engine_prelude::*;
+use crate::jni::node_rust_environment::JNINodeRustEnvironment;
+use p2p::traits::node::HighPriorityPeersStore;
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_p2p_RocksDbHighPriorityPeersStore_upsertAllHighPriorityPeers(
+    env: JNIEnv,
+    _class: JClass,
+    j_rust_global_context: JObject,
+    payload: jbyteArray,
+) -> jbyteArray {
+    jni_raw_sbor_fallible_call(&env, payload, |bytes| {
+        JNINodeRustEnvironment::get_address_book_database(&env, j_rust_global_context)
+            .upsert_all_peers(&bytes);
+        Ok(())
+    })
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_p2p_RocksDbHighPriorityPeersStore_getAllHighPriorityPeers(
+    env: JNIEnv,
+    _class: JClass,
+    j_rust_global_context: JObject,
+    payload: jbyteArray,
+) -> jbyteArray {
+    jni_sbor_coded_call(&env, payload, |_: ()| -> Option<Vec<u8>> {
+        JNINodeRustEnvironment::get_address_book_database(&env, j_rust_global_context)
+            .get_all_peers()
```

Comment:
Ditto - this looks wrong. I think perhaps this function just needs to be renamed `get_all_high_priority_peers`, or is this a bug?

Reply:
Fixed